apollo-federation/src/sources/connect/validation/selection.rs

```diff
@@ -96,13 +114,59 @@ pub(super) fn validate_body_selection(
     }
 
     // TODO: validate JSONSelection
+
+    if let Some(value) = validate_selection_variables(
+        ConnectorsContext::new(connect_coordinate.into(), Phase::Request, Target::Body),
+        schema,
+        selection_node,
+        coordinate,
+        &selection,
+    ) {
+        return value;
+    }
+
     Ok(())
 }
 
+/// Validate variable references in a JSON Selection
+fn validate_selection_variables(
+    expression_context: ConnectorsContext,
+    schema: &SchemaInfo,
+    selection_node: &Node<Value>,
+    coordinate: String,
+    selection: &JSONSelection,
+) -> Option<Result<(), Message>> {
+    let namespaces: HashSet<Namespace> = expression_context.available_namespaces().collect();
+    for reference in selection
+        .external_var_paths()
+        .into_iter()
+        .flat_map(|var_path| var_path.variable_reference())
+    {
+        if !namespaces.contains(&reference.namespace.namespace) {
+            return Some(Err(Message {
+                code: Code::InvalidJsonSelection,
+                message: format!(
+                    "{coordinate} contains an invalid variable namespace `{namespace}`, must be one of {available}",
```

Comment:
is "namespace" a term we want to expose to users? at the moment, they're documented just as ["Variables"](https://www.apollographql.com/docs/graphos/schema-design/connectors/selection#variables)